### PR TITLE
fix(adapter-pg*): fix import & test pipeline 

### PR DESCRIPTION
**Pull request by jkomyno**
This PR:
- reverts https://github.com/prisma/prisma/pull/23311
- introduces https://github.com/prisma/prisma/pull/23352 in its place (which is [already known to work on ecosystem-tests](https://github.com/prisma/prisma/pull/23352#issuecomment-1978554531))

### CODE REVIEW

**Comment by jkomyno** [comment]
https://github.com/prisma/prisma/pull/23366/commits/817feffe1d2fe93c15c84e6ab08d32ef7d1e1149 supposedly works both with `tsc` and with our `ecosystem-tests`, but it breaks Wrangler in `@prisma/bundle-size`..

File: packages/adapter-pg-worker/package.json [hunk 1]
```diff
@@ -37,14 +37,15 @@
   "sideEffects": false,
   "dependencies": {
     "@prisma/driver-adapter-utils": "workspace:*",
+    "pg-types": "^4.0.2",
```
Comment (jkomyno) [inline packages/adapter-pg-worker/package.json:40]: TODO: remove

File: packages/adapter-pg/package.json [hunk 2]
```diff
@@ -37,6 +37,7 @@
   "sideEffects": false,
   "dependencies": {
     "@prisma/driver-adapter-utils": "workspace:*",
+    "pg-types": "^4.0.2",
```
Comment (jkomyno) [inline packages/adapter-pg/package.json:40]: TODO: remove

File: packages/adapter-pg/src/conversion.ts [hunk 3]
```diff
@@ -1,9 +1,10 @@
 import { type ColumnType, ColumnTypeEnum, JsonNullMarker } from '@prisma/driver-adapter-utils'
-import * as pg from 'pg'
+// @ts-ignore: this is used to avoid the `Module '"<path>/node_modules/@types/pg/index"' has no default export.` error.
+import pg from 'pg'
 import { parse as parseArray } from 'postgres-array'
 
-const types = pg.types
-const ScalarColumnType = types.builtins
+const { types } = pg
+const { builtins: ScalarColumnType, getTypeParser, setTypeParser } = types
```
Comment (jkomyno) [inline packages/adapter-pg/src/conversion.ts:7]: This is the important change. `@prisma/adapter-pg-worker/src/conversion.ts` is basically a copy-paste of the edits to this file.

File: packages/adapter-pg-worker/src/conversion.ts [hunk 4]
```diff
@@ -2,8 +2,8 @@ import { type ColumnType, ColumnTypeEnum, JsonNullMarker } from '@prisma/driver-
 import * as pg from '@prisma/pg-worker'
 import { parse as parseArray } from 'postgres-array'
 
-const types = pg.types
-const ScalarColumnType = types.builtins
+const { types } = pg
+const { builtins: ScalarColumnType, getTypeParser, setTypeParser } = types
```
Comment (jkomyno) [inline packages/adapter-pg-worker/src/conversion.ts:6]: Move to the next comment, please

**Review by SevInf — APPROVED** [review]

**Review by Jolg42 — APPROVED** [review]
Nice 🙌🏼

**Pull request merged**